WordPress/Classes/ViewRelated/Reader/WPImageViewController.m

```diff
@@ -92,37 +122,59 @@ - (void)loadImage
     }
 
     if (self.image != nil) {
-        self.imageView.image = self.image;
-        [self.imageView sizeToFit];
-        self.scrollView.contentSize = self.imageView.image.size;
-        [self centerImage];
-
+        [self updateImageView];
     } else if (self.url) {
-        self.isLoadingImage = YES;
-        [self.activityIndicatorView startAnimating];
-        __weak __typeof__(self) weakSelf = self;
-        [_imageView setImageWithURLRequest:[NSURLRequest requestWithURL:self.url]
-                         placeholderImage:self.image
-                                  success:^(NSURLRequest *request, NSHTTPURLResponse *response, UIImage *image) {
-                                      __typeof__(self) strongSelf = weakSelf;
-                                      if (!strongSelf) {
-                                          return;
-                                      }
-                                      [strongSelf.activityIndicatorView stopAnimating];
-                                      strongSelf.imageView.image = image;
-                                      [strongSelf.imageView sizeToFit];
-                                      strongSelf.scrollView.contentSize = strongSelf.imageView.image.size;
-                                      [strongSelf centerImage];
-                                      strongSelf.isLoadingImage = NO;
-                                  } failure:^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error) {
-                                      DDLogError(@"Error loading image: %@", error);
-                                      __typeof__(self) strongSelf = weakSelf;
-                                      [strongSelf.activityIndicatorView stopAnimating];
-                                      strongSelf.isLoadingImage = NO;
-                                  }];
+        [self loadImageFromURL];
+    } else if (self.media) {
+        [self loadImageFromMedia];
     }
 }
 
+- (void)updateImageView
+{
+    self.imageView.image = self.image;
+    [self.imageView sizeToFit];
+    self.scrollView.contentSize = self.imageView.image.size;
+    [self centerImage];
+}
+
+- (void)loadImageFromURL
+{
+    self.isLoadingImage = YES;
+
+    __weak __typeof__(self) weakSelf = self;
+    [_imageView setImageWithURLRequest:[NSURLRequest requestWithURL:self.url]
+                      placeholderImage:self.image
+                               success:^(NSURLRequest *request, NSHTTPURLResponse *response, UIImage *image) {
+                                   weakSelf.image = image;
```

Comment:
I don't think it'll matter here? If `self` has been dealloc'd (the view controller has gone away), then these will all just reference `nil` and nothing will happen.